PatreonClient/Models/BaseRelationship.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using PatreonClient.Responses;
+
+namespace PatreonClient.Models
+{
+    public abstract class BaseRelationShip : IRelationship
+    {
+        protected internal BaseRelationShip AssignData<TAttributes>(IReadOnlyCollection<PatreonData> includes,
+            PatreonResponse<TAttributes> baseType)
+        {
+            if (baseType?.Data == null) return this;
+
+            baseType.Data = includes.FirstOrDefault(x => x.Id == baseType.Data.Id) as PatreonData<TAttributes>;
+
+            return this;
+        }
+
+        protected internal BaseRelationShip AssignDataAndRelationship<TAttributes, TRelationships>(
+            IReadOnlyCollection<PatreonData> includes, PatreonResponse<TAttributes, TRelationships> baseType)
+            where TRelationships : IRelationship
+        {
+            if (baseType?.Data == null) return this;
+
+            baseType.Data =
+                includes.FirstOrDefault(x => x.Id == baseType.Data.Id) as PatreonData<TAttributes, TRelationships>;
+
+            baseType.Data?.Relationships?.AssignRelationship(includes);
+
+            return this;
+        }
+
+        protected internal BaseRelationShip AssignCollectionAttributesAndRelationships<TAttributes, TRelationships>(
+            IReadOnlyCollection<PatreonData> includes,
+            PatreonCollectionResponse<TAttributes, TRelationships> baseCollection) where TRelationships : IRelationship
+        {
+            if (baseCollection?.Data == null) return this;
+
+            foreach (var collectionInstance in baseCollection.Data)
+            {
+                var include =
+                    includes.FirstOrDefault(x => x.Id == collectionInstance.Id) as
+                        PatreonData<TAttributes, TRelationships>;
+                if (include != null)
+                {
+                    collectionInstance.Attributes = include.Attributes;
+                    collectionInstance.Relationships = include.Relationships;
+                }
+
+                collectionInstance.Relationships?.AssignRelationship(includes);
+            }
+
+            return this;
+        }
+
+        public void AssignRelationship(IReadOnlyCollection<PatreonData> includes)
```

Comment:
if you make this `abstract` and don't provide the implementation you can then resolve that issue you had with not being able to override the method and having to include the interface.

`public abstract void AssignRelationship(IReadOnlyCollection<PatreonData> includes);`

Reply:
Ahah!  I thought this was possible, sometimes you just need someone to point it out to you.

With this change, it's much cleaner overall 👍